applications/GeoMechanicsApplication/custom_utilities/process_factory.cpp

```diff
@@ -0,0 +1,38 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "custom_utilities/process_factory.hpp"
+#include "includes/kratos_parameters.h"
+
+
+namespace Kratos
+{
+
+ProcessFactory::ProductType ProcessFactory::Create(const std::string& rProcessClassName,
+                                                   const Parameters&  rProcessSettings) const
+{
+    auto pos = mCreatorMap.find(rProcessClassName);
```

Comment:
This one can be const auto

Reply:
To the best of my knowledge it's rather unusual to use `const auto` for iterators (just `auto` is fine). The `const`ness is normally expressed by the kind of iterator: either an `iterator` or a `const_iterator` (they are different classes). Which one `find` returns depends on the mutability of the underlying map. For this reason, I feel it's better to not follow your suggestion.